lib/Runtime/Library/ScriptFunction.cpp

```diff
@@ -594,7 +594,7 @@ namespace Js
                     {
                         memset(this->m_inlineCaches[i], 0, sizeof(InlineCache));
                     }
-                    else if(!scriptContext->IsClosed())
+                    else if(scriptContext != nullptr && !scriptContext->IsClosed())
```

Comment:
ah, this is not intentional change. previously in order to solve the issue the when FunctionBody::Cleanup sometimes the scriptContext is already gone, I set this to null. So this is not needed any more because when cleaning up we should not be expanding the structure so should not need the recycler from scriptContext anymore.
let me remove the change. thanks.